pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -798,6 +798,17 @@ type PrometheusSpec struct {
 	// Maximum number of bytes used by the Prometheus data.
 	RetentionSize ByteSize `json:"retentionSize,omitempty"`
 
+	// ShardRetentionPolicy defines the retention policy for the Prometheus shards.
+	// (Alpha) Using this field requires the ShardRetentionPolicies feature gate to be enabled.
+	//
+	// The final goals for this feature can be seen at https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/proposals/202310-shard-autoscaling.md#graceful-scale-down-of-prometheus-servers,
+	// however, the feature is not yet fully implemented and has known limitations:
+	// * Scale up after scale down will not re-use the previously marked-for-deletion shards.
+	// * Target redistribution doesn't happen on scale up/down.
+	//
+	// +optional
+	ShardRetentionPolicy *ShardRetentionPolicy `json:"shardRetentionPolicy,omitempty"`
```

Comment:
Wdyt about not exposing this fields at all on the CRD and assume default values when the feature is enabled.

This is mainly what I've suggested in one office hours, we can assume default values for the feature, since we're underd development and we want to do increamental PRs.

So we might think better about the API and when the feature is ready to be exposed we can introduce new fields on the CRs wdyt?

Reply:
I'm a bit confused how we could implement both the retain and delete APIs with this strategy, since we plan to have multiple behaviors on retain 😬

Reply:
Sorry!
I completely missed this comment.
What I mean is, at lest in the development phases do not expose this field in the API until we reach the desired state, once we have the feature working behind the scenes we can come back and decide how we want to expose this to the API.

I particularly see two advantages.
1. We reduce the changes of people try to use it while is under development.
2. We can post pone changes to the CRD API and think it better once we have the feature working as expected.

Reply:
I'm still a bit confused 😬 How can we use the values from the API during reconciliation, but not expose it in the CRD?